tests/test_responses.py

```diff
@@ -287,6 +287,21 @@ async def send(message: Message) -> None:
     await app({"type": "http", "method": "head"}, receive, send)
 
 
+def test_file_response_with_manually_specified_media_type(
+    tmpdir: Path, test_client_factory: TestClientFactory
+) -> None:
+    path = os.path.join(tmpdir, "xyz")
+    content = b"<file content>"
+    with open(path, "wb") as file:
+        file.write(content)
+    # By default, FileResponse will determine the `media_type` based on
+    # the filename or path, unless a specific `media_type` is provided.
+    app = FileResponse(path=path, filename="example.png", media_type="image/jpeg")
+    client: TestClient = test_client_factory(app)
+    response = client.get("/")
+    assert response.headers["content-type"] == "image/jpeg"
```

Comment:
https://github.com/encode/starlette/blob/master/starlette/responses.py#L294-L295
Testing this `if` branch.